.github/workflows/rust.yml

```diff
@@ -0,0 +1,36 @@
+---
+name: "Rust CI"
+on:
+  workflow_call:
+
+jobs:
+  build:
+    name: "Build"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Prepare dependencies
+        run: |
+          rustup update stable
+          cargo install cargo-auditable
+
+      - name: Extract crate information
+        id: crate_metadata
+        run: |
+          cargo metadata --no-deps --format-version 1 | jq -r '[.packages[0].targets[] | select(.kind == ["bin"]) | "./target/release/" + .name ] | join(" ") | "binary_targets=\"" + . + "\""' | tee -a $GITHUB_OUTPUT
+
+      - name: Clippy
+        run: cargo clippy
+
+      - name: Build
+        run: cargo auditable build --release
+
+      - name: Create release
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release create ${{ github.ref_name }} ${{ steps.crate_metadata.outputs.binary_targets }}
+        if: github.ref_type == 'tag'
```

Comment:
That means every tag, is restricting to `vX.Y.z` an easy thing to do?

Reply:
not really, unless we also want the rest to not run on other tags